cmd/fs-v1.go

```diff
@@ -913,12 +914,10 @@ func (fs *FSObjects) ListObjects(ctx context.Context, bucket, prefix, marker, de
 		return loi, nil
 	}
 
-	// For delimiter and prefix as '/' we do not list anything at all
-	// since according to s3 spec we stop at the 'delimiter'
-	// along // with the prefix. On a flat namespace with 'prefix'
+	// On a flat namespace with 'prefix'
 	// as '/' we don't have any entries, since all the keys are
 	// of form 'keyName/...'
-	if delimiter == slashSeparator && prefix == slashSeparator {
+	if strings.HasPrefix(prefix, slashSeparator) {
```

Comment:
you should use the helper `hasPrefix` instead for more cross-platform compatibility.  btw, what is the issue in the previous code? we need to make sure that we return for delimiter == slashSeparator as well. also, we should only use `prefix == slashSeparator` here.